services/harvest_service.py

```diff
@@ -0,0 +1,266 @@
+
+
+class HarvestService(IHarvestService):
+    """
+    HarvestService class responsible for handling tasks such as connecting to
+    RabbitMQ and Kerberos Vault, receiving and processing messages, downloading
+    and opening video files, and processing video frames using YOLO models.
+    """
+
+    def __init__(self):
+        """
+        Constructor.
+        """
+        self.rabbitmq = None
+        self.vault = None
+        # Initialize the VariableClass object, which contains all the necessary environment variables.
+        self._var = VariableClass()
+
+    def connect(self, *agents):
+        """
+        Connects to the required agents, specifically RabbitMQ and Kerberos Vault.
+
+        Args:
+            agents (tuple): A tuple containing the names of agents to connect to.
+                            Must include 'rabbitmq' and/or 'kerberos_vault'.
+
+        Raises:
+            TypeError: If neither 'rabbitmq' nor 'kerberos_vault' is included in agents.
+        """
+        if 'rabbitmq' not in agents and 'kerberos_vault' not in agents:
+            raise TypeError('Missing agent!')
+
+        if self._var.LOGGING:
+            print('a) Initializing RabbitMQ')
+
+        # Initialize a message broker using the python_queue_reader package
+        self.rabbitmq = RabbitMQ(
+            queue_name=self._var.QUEUE_NAME,
+            target_queue_name=self._var.TARGET_QUEUE_NAME,
+            exchange=self._var.QUEUE_EXCHANGE,
+            host=self._var.QUEUE_HOST,
+            username=self._var.QUEUE_USERNAME,
+            password=self._var.QUEUE_PASSWORD)
+
+        if self._var.LOGGING:
+            print('b) Initializing Kerberos Vault')
+
+        self.vault = KerberosVault(
+            storage_uri=self._var.STORAGE_URI,
+            storage_access_key=self._var.STORAGE_ACCESS_KEY,
+            storage_secret_key=self._var.STORAGE_SECRET_KEY)
+
+    def receive_message(self):
+        """
+        Receives a message from RabbitMQ and retrieves the corresponding media
+        from Kerberos Vault.
+
+        Returns:
+            dict or None: The received message if available, otherwise None.
+        """
+        # Receive message from the queue,
+        # and retrieve the media from the Kerberos Vault utilizing the message information.
+        if self._var.LOGGING:
+            print('1) Receiving message from RabbitMQ')
+        message = self.rabbitmq.receive_message()
+        if not message:
+            if self._var.LOGGING:
+                print('No message received, waiting for 3 seconds')
+            time.sleep(3)
+            return None
+        if self._var.LOGGING:
+            print('2) Retrieving media from Kerberos Vault')
+
+        self.vault.retrieve_media(
+            message=message,
+            media_type='video',
+            media_savepath=self._var.MEDIA_SAVEPATH)
+        return message
+
+    def process_from_vault(self, media_key, provider):
```

Comment:
This is a confusing name. The process from vault, removes the recording from kerberos vault. So I would expect we call it `delete_media`?

Ideally this function should also be moved to the dedicated repo: https://github.com/uug-ai/uugai-python-kerberos-vault/blob/main/uugai_python_kerberos_vault/KerberosVault.py